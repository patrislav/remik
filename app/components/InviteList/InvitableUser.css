.wrapper {
  border: none;
  padding: 0.3rem;
  background-color: inherit;
  width: 100%;
  display: flex;
  align-items: center;
  transition: all 250ms ease-in-out;
}

.wrapper:hover {
  background-color: #eee;
}

.photo {
  width: 36px;
  height: 36px;
  border-radius: 50%;
}

.name {
  margin-left: 0.3rem;
}
