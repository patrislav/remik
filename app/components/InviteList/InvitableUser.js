import React, {PropTypes} from 'react'
import styles from './InvitableUser.css'

const propTypes = {
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    photoURL: PropTypes.string.isRequired
  }).isRequired
}

const InvitableUser = ({ user, ...props }) => (
  <button className={styles.wrapper} {...props}>
    <img src={user.photoURL} className={styles.photo} />
    <span className={styles.name}>{user.name}</span>
  </button>
)

InvitableUser.propTypes = propTypes

export default InvitableUser
