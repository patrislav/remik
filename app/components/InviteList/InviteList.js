import React, {PropTypes} from 'react'
import {connect} from '../../lib/rxstate'
import {selectInvitableUsers} from '../../modules/users/selectors'
import styles from './InviteList.css'

import InvitableUser from './InvitableUser'

const propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    photoURL: PropTypes.string
  })),
  onInvite: PropTypes.func
}

const defaultProps = {
  users: [],
  onInvite: () => {}
}

export const InviteList = ({ users, onInvite }) => (
  <section className={styles.section}>
    <ul className={styles.list}>
      {users.map(user =>
        <li key={user.id} className={styles.item}>
          <InvitableUser user={user} onClick={() => onInvite(user.id)} />
        </li>
      )}
    </ul>
  </section>
)

InviteList.propTypes = propTypes
InviteList.defaultProps = defaultProps

export default connect(state => ({
  users: selectInvitableUsers(state)
}))(InviteList)
