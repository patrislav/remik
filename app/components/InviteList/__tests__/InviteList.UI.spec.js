import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import times from 'lodash/times'

import {InviteList} from '../InviteList'

describe('<InviteList />', () => {
  it('renders correct UI without props', () => {
    const wrapper = shallow(<InviteList />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correct UI with passed users', () => {
    const users = times(3, i => ({
      id: i.toString(),
      name: `User #${i}`,
      photoURL: `PHOTO_URL_#${i}`
    }))
    const wrapper = shallow(<InviteList users={users} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
