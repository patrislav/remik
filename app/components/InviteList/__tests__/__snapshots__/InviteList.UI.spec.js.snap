exports[`<InviteList /> renders correct UI with passed users 1`] = `
<section
  className="section">
  <ul
    className="list">
    <li
      className="item">
      <InvitableUser
        onClick={[Function]}
        user={
          Object {
            "id": "0",
            "name": "User #0",
            "photoURL": "PHOTO_URL_#0",
          }
        } />
    </li>
    <li
      className="item">
      <InvitableUser
        onClick={[Function]}
        user={
          Object {
            "id": "1",
            "name": "User #1",
            "photoURL": "PHOTO_URL_#1",
          }
        } />
    </li>
    <li
      className="item">
      <InvitableUser
        onClick={[Function]}
        user={
          Object {
            "id": "2",
            "name": "User #2",
            "photoURL": "PHOTO_URL_#2",
          }
        } />
    </li>
  </ul>
</section>
`;

exports[`<InviteList /> renders correct UI without props 1`] = `
<section
  className="section">
  <ul
    className="list" />
</section>
`;
